packages/investor-foxy/src/api/api.ts

```diff
@@ -232,99 +230,155 @@ export class FoxyApi {
     const stakingContract = this.getStakingContract(addresses.staking)
 
     try {
-      const expired = await stakingContract.methods.pauseStaking().call()
+      const expired = await stakingContract.pauseStaking()
       const tvl = await this.tvl({ tokenContractAddress: addresses.foxy })
       const apy = this.apy()
       return transformData({ ...addresses, tvl, apy, expired })
-    } catch (err) {
-      throw new Error(`Failed to get contract data ${err}`)
+    } catch (e) {
+      throw new Error(`Failed to get contract data ${e}`)
     }
   }
 
-  async getGasPrice() {
-    const gasPrice = await this.web3.eth.getGasPrice()
-    return bnOrZero(gasPrice)
-  }
-
-  getTxReceipt({ txid }: TxReceipt): Promise<TransactionReceipt> {
+  getTxReceipt({ txid }: TxReceipt): Promise<ethers.providers.TransactionReceipt> {
     if (!txid) throw new Error('Must pass txid')
-    return this.web3.eth.getTransactionReceipt(txid)
+    return this.provider.getTransactionReceipt(txid)
   }
 
-  async estimateClaimWithdrawGas(input: ClaimWithdrawal): Promise<BigNumber> {
+  async estimateClaimWithdrawFees(
+    input: ClaimWithdrawal,
+  ): Promise<FeeDataEstimate<KnownChainIds.EthereumMainnet>> {
     const { claimAddress, userAddress, contractAddress } = input
     const addressToClaim = claimAddress ?? userAddress
     this.verifyAddresses([addressToClaim, userAddress, contractAddress])
 
     const stakingContract = this.getStakingContract(contractAddress)
 
     try {
-      const estimatedGas = await stakingContract.methods.claimWithdraw(addressToClaim).estimateGas({
-        from: userAddress,
+      const data = stakingContract.interface.encodeFunctionData('claimWithdraw', [addressToClaim])
+      const estimatedFees = await this.adapter.getFeeData({
```

Comment:
and just to state it, all of my updates used `feeData` as this return value var name

Reply:
ed53a49d7545a150000e23f5e1a78c4c2f7ecda4